src/lib/modernExtend.ts

```diff
@@ -196,6 +196,10 @@ export function setupConfigureForReading(cluster: string | number, attributes: (
     return configure;
 }
 
+export function autoDetectInputEndpoint(device: Zh.Device, cluster: string | number): Zh.Endpoint | Zh.Group {
+    return device.endpoints.find((e) => e.supportsInputCluster(cluster));
```

Comment:
```suggestion
    return device.endpoints.find((e) => e.supportsInputCluster(cluster)) ?? device.endpoints[0];
```

When non of the endpoints support it, which should try reading it from the first (not all devices might discover their clusters correctly)